lib/puppet/provider/group/gpasswd.rb

```diff
@@ -0,0 +1,75 @@
+require 'puppet/provider/group/groupadd'
+
+Puppet::Type.type(:group).provide :gpasswd, :parent => Puppet::Type::Group::ProviderGroupadd do
+  require 'set'
+  require 'shellwords'
+
+  desc <<-EOM
+    Group management via `gpasswd`. This allows for local group
+    management when the users exist in a remote system.
+  EOM
+
+  commands  :addmember => 'gpasswd',
+            :delmember => 'gpasswd'
+
+  has_feature :manages_members unless %w{HP-UX Solaris}.include? Facter.value(:operatingsystem)
```

Comment:
Is having the `gpasswd` command sufficient to be able to manage group membership? If so, we could remove the `operatingsystem` check.

If not, are we missing some systems to exclude like AIX? Perhaps it would be better to specify a whitelist of platforms it's known to work on?